dash-spv-coinjoin/src/messages/coinjoin_status_update.rs

```diff
@@ -0,0 +1,60 @@
+use std::io;
+use std::io::{Error, Write};
+use dash_spv_masternode_processor::consensus::{Decodable, Encodable, encode};
+
+use super::pool_message::PoolMessage;
+use super::pool_state::PoolState;
+use super::pool_status_update::PoolStatusUpdate;
+
+// dssu
+#[repr(C)]
+#[derive(Clone, Debug)]
+#[ferment_macro::export]
+pub struct CoinJoinStatusUpdate {
+    pub session_id: i32,
+    pub pool_state: PoolState,
+    pub status_update: PoolStatusUpdate,
+    pub message_id: PoolMessage,
+}
+
+impl Encodable for CoinJoinStatusUpdate {
+    #[inline]
+    fn consensus_encode<W: Write>(&self, mut writer: W) -> Result<usize, Error> {
+        let mut offset = 0;
+        offset += (self.session_id as u32).consensus_encode(&mut writer)?;
+        offset += (self.pool_state.value() as u32).consensus_encode(&mut writer)?;
+
+        // TODO: versioning
+        // if (protocolVersion <= params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.COINJOIN_SU)) {
+        //     Utils.uint32ToByteStreamLE(0, stream); // nEntriesCount, deprecated
+        // }
```

Comment:
Since we don't need to interact with old versions of DashCore, eventually, this won't be required at all.